### PR TITLE
refactor(connector): replace hyper client implementation with reqwest

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

`hyper::Client` is deprecated in `hyper v1` because it's considered to be [higher-level](https://arc.net/l/quote/ckkcpvun). 
We'd better switch to `reqwest` for high-level HTTP client.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/16146?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#16146** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/16146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#16145** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/16145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BugenZhao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/connector/src/sink/clickhouse.rs [hunk 1]
```diff
@@ -190,18 +190,9 @@ impl ClickHouseEngine {
     }
 }
 
-const POOL_IDLE_TIMEOUT: Duration = Duration::from_secs(5);
```
Comment (wenym1) [inline src/connector/src/sink/clickhouse.rs:193]: Is is ok to remove this `POOL_IDLE_TIMEOUT` setting? I guess the original purpose not to use `ClickHouseClient::default()` is to set this config. cc @xxhZs

**Review by wenym1 — COMMENTED** [review]
@xxhZs PTAL

Can you run sink benchmark on the related sinks to see if the change affect the sink performance? You may ask @xxhZs for details about the sink benchmark.

File: src/connector/src/sink/clickhouse.rs (hunk 1 above)
Comment (xxhZs) [reply src/connector/src/sink/clickhouse.rs:193]: The corresponding pr is this one, https://github.com/risingwavelabs/risingwave/pull/12041/files
 the default value of `POOL_IDLE_TIMEOUT` is 2s in clickhouse.rs,  the reason for not using the default here is to support https urls

Comment (wenym1) [reply src/connector/src/sink/clickhouse.rs:193]: If so, we may not modify the code here? @BugenZhao

Comment (wenym1) [reply src/connector/src/sink/clickhouse.rs:193]: May need to run a test to connect to clickhouse with https. If it passes, we can then sync the change.

Comment (BugenZhao) [reply src/connector/src/sink/clickhouse.rs:193]: Cherry-picked the fix into our fork of `clickhouse.rs`.

**Review by xxhZs — APPROVED** [review]
lgtm

File: src/connector/src/sink/doris_starrocks_connector.rs [hunk 2]
```diff
@@ -187,33 +186,27 @@ impl InserterInnerBuilder {
...
-            .body(Body::empty())
-            .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
-        let resp = client
-            .request(request_get_url)
+        let builder = self.build_request(self.url.clone());
+        let resp = builder
+            .send()
             .await
             .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+        // TODO: shall we let `reqwest` handle the redirect?
```
Comment (xxhZs) [inline src/connector/src/sink/doris_starrocks_connector.rs:209]: Sounds good, I wonder if the logic of the following special verdict is an obstacle cc @xuefengze

**Comment by BugenZhao** [comment]
#15429 introduced another occurrence. 🤡🤡

**Review by wenym1 — APPROVED** [review]

**Pull request merged**